npbench/benchmarks/nbody/nbody_triton.py

```diff
@@ -0,0 +1,272 @@
+    mj_2d = mj[None, :]      # [1, BLOCK_SIZE_K]
+    mm = mi_2d * mj_2d       # [BLOCK_SIZE_N, BLOCK_SIZE_K]
+
+    # energy contribution per pair: -G * m_i m_j / r_ij
+    tile_energy = -G * mm * inv_r
+
+    # Sum over this tile
+    tile_sum = tl.sum(tile_energy, axis=0)
+    tile_sum = tl.sum(tile_sum, axis=0)  # scalar
+
+    # Atomically add into global accumulator
+    tl.atomic_add(pe, tile_sum)
+   
+
+def nbody(mass, pos, vel, N, Nt, dt, G, softening):
+    """
+    Calculate the acceleration on each particle due to Newton's Law 
+    pos  is an N x 3 matrix of positions
+    mass is an N x 1 vector of masses
+    G is Newton's Gravitational constant
+    softening is the softening length
+    a is N x 3 matrix of accelerations
+    vel is N x 3 matrix of velocities
+    """
+    # Make sure N, Nt, G, softening, dt are plain Python scalars
+    N = int(N)
+    Nt = int(Nt)
+    G = float(G)
+    softening = float(softening)
+    dt = float(dt)
+
+    # Get DTYPE and assert dtypes
+    assert mass.dtype == pos.dtype == vel.dtype, "mass, pos, and vel must have the same dtype"
+    dtype = pos.dtype
+    assert dtype in (torch.float32, torch.float64)
+    DTYPE = tl.float32 if dtype == torch.float32 else tl.float64
+
+    # define grids for kernel launches
+    grid_1d = lambda meta: (triton.cdiv(N, meta["BLOCK_SIZE_N"]),)
+    grid_2d = lambda meta: (triton.cdiv(N, meta["BLOCK_SIZE_N"]),
+                            triton.cdiv(N, meta["BLOCK_SIZE_K"]))
+
+    # Convert to Center-of-Mass frame
+    # vel -= np.mean(mass * vel, axis=0) / np.mean(mass)
+    mom = (mass * vel).mean(dim=0)   # shape (3,)
+    m_mean = mass.mean()             # scalar
+
+    vel -= mom / m_mean
+
+    # calculate initial gravitational accelerations
+    # acc = getAcc(pos, mass, G, softening)
+    acc = torch.zeros((N, 3), dtype=pos.dtype)
+    _get_acc[grid_1d](pos, mass, G, softening, acc, N, DTYPE)
+
+    # # calculate initial energy of system
+    # KE = np.ndarray(Nt + 1, dtype=mass.dtype)
+    # PE = np.ndarray(Nt + 1, dtype=mass.dtype)
+    # KE[0], PE[0] = getEnergy(pos, vel, mass, G)
+    KE = torch.empty(Nt + 1, dtype = dtype)
+    PE = torch.empty(Nt + 1, dtype = dtype)
+    pe_acc = torch.zeros((1,), dtype=dtype)
+    _get_energy[grid_2d](pos, mass, G, pe_acc, N, DTYPE)
+    KE[0] = 0.5 * torch.sum(mass * vel**2) 
+    PE[0] = pe_acc[0]
+
+    # Main loop
+    t = 0.0
+    for i in range(Nt):
+        # 1/2 kick
+        vel += acc * (dt / 2.0)
+
+        # drift
+        pos += vel * dt
+
+        # update accelerations
+        _get_acc[grid_1d](pos, mass, G, softening, acc, N, DTYPE)
+
+        # 1/2 kick
+        vel += acc * (dt / 2.0)
+        t += dt
+
+        # get energy of system
+        pe_acc.zero_()
+        _get_energy[grid_2d](pos, mass, G, pe_acc, N, DTYPE)
+        KE[i + 1] = 0.5 * torch.sum(mass * vel**2)
+        PE[i + 1] = pe_acc[0]
+
+    return KE.cpu().numpy(), PE.cpu().numpy()
```

Comment:
```suggestion
    return KE, PE
```
This should not be necessary and may even harm performance